src/bosh-director/lib/bosh/director/deployment_plan/instance.rb

```diff
@@ -187,8 +187,38 @@ def update_instance_settings
           { 'name' => disk.model.name, 'cid' => disk.model.disk_cid }
         end
 
-        agent_client.update_settings(Config.trusted_certs, disk_associations)
-        @model.active_vm.update(trusted_certs_sha1: ::Digest::SHA1.hexdigest(Config.trusted_certs))
+        settings = {
+          'trusted_certs' => Config.trusted_certs,
+          'disk_associations' => disk_associations,
+        }
+        if blobstore_config_changed?
+          blobstore = App.instance.blobstores.blobstore
+          blobstore.validate!(Config.agent_env['blobstores'].first.fetch('options', {}), @stemcell.api_version)
+          if blobstore.can_sign_urls?(@stemcell.api_version)
+            settings['blobstores'] = blobstore.redact_credentials(Config.agent_env['blobstores'])
+          else
+            settings['blobstores'] = Config.agent_env['blobstores']
+          end
+        end
+
+        if nats_config_changed?
+          cert_generator = NatsClientCertGenerator.new(@logger)
+          agent_cert_key_result = cert_generator.generate_nats_client_certificate "#{vm.agent_id}.agent.bosh-internal"
+          settings['mbus'] = {
+            'cert' => {
+              'ca' => Config.nats_server_ca,
+              'certificate' => agent_cert_key_result[:cert].to_pem,
+              'private_key' => agent_cert_key_result[:key].to_pem,
+            }
+          }
+        end
+
+        AgentClient.with_agent_id(vm.agent_id, @model.name).update_settings(settings)
```

Comment:
In this class `agent_client.*` is used only this place doesn't use it anymore. Maybe we should add a comment to make sure why it doesn't use `agent_client.*`.